project essentials

ajantha-bhat opened the pull request:
Just copied the project essentials like License, gitignore, code style from https://github.com/projectnessie/nessie-apprunner

XN137 commented on codestyle/checkstyle-config.xml:
```diff
@@ -0,0 +1,177 @@
+<?xml version="1.0"?>
+<!--
+  ~ Copyright (C) 2021 Dremio
```
2022

ajantha-bhat commented on codestyle/eclipse-java-google-style.xml:
```diff
@@ -0,0 +1,337 @@
+<setting id="org.eclipse.jdt.core.formatter.insert_space_before_binary_operator" value="insert"/>
+<setting id="org.eclipse.jdt.core.formatter.blank_lines_before_package" value="0"/>
+<setting id="org.eclipse.jdt.core.formatter.alignment_for_cascading_method_invocation_with_arguments" value="16"/>
+<setting id="org.eclipse.jdt.core.compiler.source" value="1.7"/>
```
as mentioned in the PR description, it is same as https://github.com/projectnessie/nessie-apprunner

I think it is harmless to keep.

dimas-b replied:
Do we enforce code format with `spotless`?

ajantha-bhat replied:
I just removed eclipse-related stuff and kept only IntelliJ stuff. I think IDE picks this file automatically if exists. 

I will enforce spotless once I have pom files (in the next PR).

ajantha-bhat commented on NOTICE:
```diff
@@ -0,0 +1,2 @@
+Dremio
+Copyright 2015-2017 Dremio Corporation
```
I have no idea. Other projects have the same.

dimas-b replied:
Let's use `2022`... 2015-17 just does not make sense for this fresh project.

ajantha-bhat replied:
ok. kept it just as 2022

dimas-b commented on codestyle/org.eclipse.wst.xml.core.prefs:
```diff
@@ -0,0 +1,7 @@
+eclipse.preferences.version=1
```
I'm not sure of the value of these boiler plate files. I suppose if we get contributors using Eclipse they can open PRs to add Eclipse configs... otherwise this looks like dead code.

ajantha-bhat replied:
removed

dimas-b commented on codestyle/intellij-java-google-style.xml:
```diff
@@ -0,0 +1,598 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<code_scheme name="GoogleStyle">
```
Does anyone use these files? I do not think I use this in my Nessie project :thinking: IJ is generally pretty smart to figure out the code style from existing files and Nessie forces the stype via `spotless`.

I think we should just use `splotless` + google style from day 1 in this repo. WDYT?

ajantha-bhat replied:
I just removed eclipse-related stuff and kept only IntelliJ stuff. I think IDE picks this file automatically if exists. 

I will enforce spotless once I have pom files (in the next PR).

ajantha-bhat commented:
@dimas-b, @XN137: I have addressed the comments. I will enforce the spotless once I have the POM file (next PR).

I think PR is good to go. Please recheck and approve.

XN137 reviewed (approved):
thanks for the updates, lgtm

ajantha-bhat commented:
BTW, I have not added the branch protection rule and approver list for this repo yet. Because I don't have for permission to see how it is configured for other projects in projectNessie. 
Maybe @snazy can add those for this project and dethrone from admin rights :)
